bundles/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/connection/OneWireConnection.java

```diff
@@ -81,91 +94,155 @@ public static synchronized OwfsConnection getConnection() {
+            } else {
                 cvIsEstablished = false;
-                return false;
             }
-        } else {
-            logger.warn("Couldn't connect to owserver because of missing connection parameters [IP '{}' Port '{}'].",
-                    cvIp, cvPort);
+        } catch (OwfsException oe) {
+            logger.warn("Unexpected owfs exception: {}", oe.getMessage(), oe);
+        } catch (IOException e) {
+            logger.warn("Unexpected connection failure.", e);
+        }
+
+        while (!connected && retriesRemaining > 0) {
+            logger.warn("Connection failed. Will retry in {} seconds.", cvServerRetryInterval);
+            synchronized (cvOwConnection) {
+                try {
+                    cvOwConnection.wait(cvServerRetryInterval * 1000L);
+                } catch (InterruptedException e) {
+                    logger.debug("Wait was interrupted.");
+                }
+            }
+            attempts++;
+            retriesRemaining--;
+            logger.info("Retrying failed connection... Attempt {} of {}.", attempts, cvServerRetries);
+            try {
+                result = cvOwConnection.listDirectory("/");
+                if (result != null) {
+                    connected = true;
+                }
+            } catch (OwfsException oe) {
+                logger.warn("Unexpected owfs exception: {}", oe.getMessage(), oe);
+            } catch (IOException e) {
+                logger.warn("Unexpected connection failure.", e);
+            }
+        }
+
+        if (!connected) {
+            logger.error("Couldn't connect to owserver [IP '{}' Port '{}']", cvIp, cvPort);
+            cvIsEstablished = false;
             return false;
         }
+
+        logger.info("Connected to owserver [IP '{}' Port '{}']", cvIp, cvPort);
+        cvIsEstablished = true;
+        return true;
     }
 
     /**
      * Reconnects to owserver
-     * 
+     *
      * @return
      */
     public static synchronized boolean reconnect() {
         logger.info("Trying to reconnect to owserver...");
         try {
             cvOwConnection.disconnect();
         } catch (Exception lvException) {
-            logger.error("Error while disconnecting from owserver: " + lvException, lvException);
+            logger.error("Error while disconnecting from owserver: ", lvException);
         }
         cvOwConnection = null;
+        cvIsEstablished = false;
         return connect();
     }
 
     public static synchronized void updated(Dictionary<String, ?> pvConfig) throws ConfigurationException {
-        if (pvConfig != null) {
-            logger.debug("OneWire configuration present. Setting up owserver connection.");
-            cvIp = (String) pvConfig.get("ip");
+        if (pvConfig == null) {
+            logger.info(
+                    "OneWireBinding configuration is not present. Please check your configuration file or if not needed remove the OneWireBinding addon.");

```

Comment:
It is "normal" for `updated` to be called with a `null` argument in certain situations, in which case the INFO log message will be irritating to the user.  I recommend leaving it at DEBUG level.